### PR TITLE
doc: Document behaviour of authz.enforceForSiteAdmins with p4 permissions

### PR DESCRIPTION
**Pull request by indradhanush**
## Test plan

Doc only

### CODE REVIEW

File: doc/admin/repo/perforce.md [hunk 1]
```diff
@@ -182,6 +182,7 @@ To enable file-level permissions:
 - As a special case, commits in which a user does not have permissions to read any files are hidden. If a user can read a subset of files in a commit, only those files are shown.
 - [The host field from protections are not supported](#known-issues-and-limitations).
 - [file-level permissions must be disabled for Batch Changes to work](#known-issues-and-limitations).
+- Setting `authz.enforceForSiteAdmins` to `true` in the site configuration will enforce permissions for admin users as well and they may not be able to see repositories and their contents based on their Sourcegraph user account email if it does not match with their email on the perforce server.
```
Comment (sashaostrikov) [inline doc/admin/repo/perforce.md:185]: The longest sentence in the universe 😂 
Can you pls split it so that it can be ingested more easily?

**Review by sashaostrikov — COMMENTED** [review]
Let's split the sentence, it is quite long :)

File: doc/admin/repo/perforce.md (hunk 1 above)
Comment (indradhanush) [inline doc/admin/repo/perforce.md:185]:
```suggestion
- Setting `authz.enforceForSiteAdmins` to `true` in the site configuration will enforce permissions for admin users. They may not be able to see repositories and their contents if their Sourcegraph user account email does not match with their email on the perforce server.
```

File: doc/admin/repo/perforce.md [hunk 2]
```diff
@@ -182,6 +182,7 @@ To enable file-level permissions:
 - As a special case, commits in which a user does not have permissions to read any files are hidden. If a user can read a subset of files in a commit, only those files are shown.
 - [The host field from protections are not supported](#known-issues-and-limitations).
 - [file-level permissions must be disabled for Batch Changes to work](#known-issues-and-limitations).
+- Setting `authz.enforceForSiteAdmins` to `true` in the site configuration will enforce permissions for admin users. They may not be able to see repositories and their contents if their Sourcegraph user account email does not match with their email on the perforce server.
```
Comment (sashaostrikov) [inline doc/admin/repo/perforce.md:185]: From what I see in the docs, Perforce is capitalized.

```suggestion
- Setting `authz.enforceForSiteAdmins` to `true` in the site configuration will enforce permissions for admin users. They may not be able to see repositories and their contents if their Sourcegraph user account email does not match with their email on the Perforce server.
```

**Review by sashaostrikov — APPROVED** [review]
Added a comment, approved to unblock 👍🏻

**Pull request merged**